flash-lso/tests/integration_tests.rs

```diff
@@ -19,6 +19,36 @@ impl<'a> fmt::Debug for PrettyArray<'a> {
     }
 }
 
+macro_rules! packet_test {
+    ($([$name: ident, $path: expr, $exact_lengths: expr]),*) => {
+        $(
+        #[test]
+        pub fn $name() -> Result<(), Box<dyn std::error::Error>> {
+            let data = include_bytes!(concat!("packet/", $path, ".dat"));
+            let parse_res = flash_lso::packet::read::parse_incomplete(data);
+
+            if let Ok((unparsed_bytes, mut packet)) =  parse_res {
+                println!("{:#?}", packet);
+
+                let empty: Vec<u8> = vec![];
+                if unparsed_bytes.len() > 0 {
+                    assert_eq!(crate::PrettyArray(&empty), crate::PrettyArray(&unparsed_bytes[..unparsed_bytes.len().min(100)].to_vec()));
+                }
+
+                let bytes = flash_lso::packet::write::write_to_bytes(&mut packet, $exact_lengths)?;
+
+                assert_eq!(crate::PrettyArray(&bytes), crate::PrettyArray(&data.to_vec()), "library output != input");
+            } else {
+                println!("parse failed: {:?}", parse_res);
```

Comment:
Csn this just be a panic?

Reply:
It's a carbon copy of the `auto_test` macro, just with a different directory and a new boolean. It does seem to panic on the next line.

I agree that that makes sense, though! I can clean it up if you think it's important for this PR?

Reply:
That's fine - it can be done later